Mathlib/NumberTheory/Dioph.lean

```diff
@@ -439,8 +439,8 @@ theorem diophFn_vec_comp1 {S : Set (Vector3 ℕ (succ n))} (d : Dioph S) {f : Ve
     (df : DiophFn f) : Dioph {v : Vector3 ℕ n | (f v::v) ∈ S} :=
   Dioph.ext (diophFn_comp1 (reindex_dioph _ (none::some) d) df) (fun v => by
     dsimp
-    -- Porting note: `congr` used to be enough here
-    suffices ((f v ::ₒ v) ∘ none :: some) = f v :: v by rw [this]; rfl
+    apply iff_of_eq
+    congr 1
```

Comment:
Can `congr!` handle this without `iff_of_eq`?

Reply:
(and below)

Reply:
`congr!` does not do anything here, but it does work a few places below. Not sure why!